src/schema/v2/home/home_page_fairs_module.ts

```diff
@@ -19,32 +19,38 @@ export const HomePageFairsModuleType = new GraphQLObjectType<
           sort: "-start_at",
           active: true,
         }
+
+        const now = moment.utc()
+
         return fairsLoader(gravityOptions).then(
-          ({ body: ungroupedRunningFairs }) => {
+          ({ body: unfilteredRunningFairs }) => {
             // Gravity returns fairs that are both current and upcoming.
-            // Make sure the current ones appear first in the results list.
-            const now = moment.utc()
-            const returnValue = groupBy(ungroupedRunningFairs, (fair) => {
+            // Make sure only the current ones appear in the results list.
+            const runningFairs = filter(unfilteredRunningFairs, (fair) => {
               const startAt = moment.utc(fair.start_at)
-              return now.isAfter(startAt) ? "current" : "upcoming"
+              return now.isAfter(startAt)
             })
 
-            const runningFairs = (returnValue["current"] || []).concat(
-              returnValue["upcoming"] || []
-            )
-
             // If there are less than 8, get the most recent closed fairs
             if (runningFairs.length >= 8) {
               return runningFairs
             }
+
             const newOptions = {
               ...gravityOptions,
               status: "closed",
               active: false,
               size: 8 - runningFairs.length,
             }
-            return fairsLoader(newOptions).then(({ body: closedFairs }) =>
-              runningFairs.concat(closedFairs)
+
+            return fairsLoader(newOptions).then(
+              ({ body: unfilteredClosedFairs }) => {
+                const closedFairs = filter(unfilteredClosedFairs, (fair) => {
+                  const endAt = moment.utc(fair.end_at)
+                  return now.isAfter(endAt)
```

Comment:
Minor and probably more a question for my understanding, but just to be sure: 

Does `closed` fairs include fairs that aren't in the past? I'm basically guessing what "closed" means, but it is surprising to me that we'd also have to filter out the ones that are closed *and* not in the past.

Reply:
That's the same question I had. During my tests, the query loader seemed to give me more than I ask for. The first query asks for running and gets running and upcoming. The second asks for closed and gets closed and upcoming. I don't know if that's a gravity issue/particularity, but I had to filter on the closed query too :(